src/chocolatey.tests/infrastructure.app/commands/ChocolateyHelpCommandSpecs.cs

```diff
@@ -0,0 +1,128 @@
+    using chocolatey.infrastructure.app.services;
+    using chocolatey.infrastructure.commandline;
+    using Moq;
+    using Should;
+
+    public class ChocolateyHelpCommandSpecs
+    {
+        public abstract class ChocolateyHelpCommandSpecsBase : TinySpec
+        {
+            protected ChocolateyHelpCommand command;
+            protected Mock<IChocolateyPackageService> packageService = new Mock<IChocolateyPackageService>();
+            protected ChocolateyConfiguration configuration = new ChocolateyConfiguration();
+
+            public override void Context()
+            {
+                configuration.Sources = "bob";
+                command = new ChocolateyHelpCommand(packageService.Object);
+            }
+        }
+
+        public class when_implementing_command_for : ChocolateyHelpCommandSpecsBase
+        {
+            private List<string> results;
+
+            public override void Because()
+            {
+                results = command.GetType().GetCustomAttributes(typeof(CommandForAttribute), false).Cast<CommandForAttribute>().Select(a => a.CommandName).ToList();
+            }
+
+            [Fact]
+            public void should_implement_help()
+            {
+                results.ShouldContain("help");
+            }
+
+        }
+
+        public class when_configurating_the_argument_parser : ChocolateyHelpCommandSpecsBase
+        {
+            private OptionSet optionSet;
+
+            public override void Context()
+            {
+                base.Context();
+                optionSet = new OptionSet();
+            }
+
+            public override void Because()
+            {
+                command.configure_argument_parser(optionSet, configuration);
+            }
+
+        }
+
+        public class when_handling_additional_argument_parsing : ChocolateyHelpCommandSpecsBase
+        {
+            private readonly IList<string> unparsedArgs = new List<string>();
+            private readonly string source = "https://somewhereoutthere";
+            private Action because;
+
+            public override void Context()
+            {
+                base.Context();
+                unparsedArgs.Add("pkg1");
+                unparsedArgs.Add("pkg2");
+                configuration.Sources = source;
+            }
+
+            public override void Because()
+            {
+                because = () => command.handle_additional_argument_parsing(unparsedArgs, configuration);
+            }
+
+            [Fact]
+            public void should_set_unparsed_arguments_to_configuration_input()
+            {
+                because();
+                configuration.Input.ShouldEqual("pkg1 pkg2");
+            }
+
+            [Fact]
+            public void should_leave_source_as_set()
+            {
+                configuration.ListCommand.LocalOnly = false;
+                because();
+                configuration.Sources.ShouldEqual(source);
+            }
+        }

```

Comment:
Yeah, you likely don't need any of this testing for when_handling_additional_argument_parsing